enzy_htp/_interface/rosetta_interface.py

```diff
@@ -1499,4 +1499,129 @@ def build_cartesian_ddg_engine(
+            if not rr.is_canonical(): 
+                continue
+            existing.append( rr.create_seq_res() )        
+        
+        seq_res = missing_residues + existing
+        seq_res.sort()
+        
+        existing = 1
+        content = list()
+        
+        for sidx, sr in enumerate(seq_res):
+        
+            if sr.missing:
+                line = f"0 X L PIKAA {sr.one_letter()}"
+            else:
+                borders_loop:bool = (
+                    sidx > 0 and seq_res[sidx-1].missing 
+                ) or (
+                    sidx < len(seq_res) - 1 and seq_res[sidx+1].missing                
+                )
+                if borders_loop:
+                    line = f"{existing} {sr.one_letter()} L PIKAA {sr.one_letter()}"
+                else:
+                    line = f"{existing} {sr.one_letter()} ."
+                existing += 1
+            
+            content.append(line)
+        
+        remodel_file:str = f"{work_dir}/input.remodel"
+        fs.write_lines( remodel_file, content ) 
+        lines=f"""<ROSETTASCRIPTS>
+<SCOREFXNS>
+</SCOREFXNS>
+<RESIDUE_SELECTORS>
+</RESIDUE_SELECTORS>
+<PACKER_PALETTES>
+</PACKER_PALETTES>
+<TASKOPERATIONS>
+</TASKOPERATIONS>
+<MOVE_MAP_FACTORIES>
+</MOVE_MAP_FACTORIES>
+<SIMPLE_METRICS>
+</SIMPLE_METRICS>
+<FILTERS>
+</FILTERS>
+<MOVERS>
+    <RemodelMover name="rm" blueprint="{remodel_file}" />
+</MOVERS>
+<PROTOCOLS>
+    <Add mover_name="rm" />
+</PROTOCOLS>
+<OUTPUT />
+</ROSETTASCRIPTS>""".splitlines()
+        script_file:str=f"{work_dir}/remodel_script.xml"
+        pdb_file = f"{work_dir}/remodel_input.pdb"
+        pdb_outfile = f"{work_dir}/remodel_input_0001.pdb"
+        
+        fs.write_lines( script_file, lines )
+        
+        opts:List[str] = ["-in:file:s", pdb_file,
+                "-parser:protocol", script_file,
+                "-remodel:quick_and_dirty",
+                "-remodel:num_trajectory", "1",
+                "-out:path:all", f"{work_dir}",
+                "-overwrite"]
+        
+        parser = PDBParser() 
+        parser.save_structure( pdb_file, aa_stru )
+        self.run_rosetta_scripts( opts )
+        
+        if not Path(pdb_outfile).exists():
+            err_msg:str=f"The output file '{pdb_outfile}' was not created. Error during loop reconstruction."
+            _LOGGER.error( err_msg )
+            raise ValueError( err_msg )
+        
+        filled_stru = parser.get_structure(pdb_outfile)
+        
+        for sr, res in zip(seq_res, filter(lambda rr: rr.is_canonical(), filled_stru.residues)):
+            res.idx = sr.idx
+            res.parent.name = sr.chain 
+        
+        for r1, r2 in zip(aa_stru.residues, filled_stru.residues):
+            r2.parent.name = r1.parent.name
+        
+        filled_stru.chains = filled_stru.chains + nc_chains
+        translate_structure(filled_stru, start_naming='rosetta' )
+        stru.chains = filled_stru.chains
```

Comment:
This have risk when a `Chain/Residue/Atom` in `stru` is already referenced by some other code

Reply:
We can leave a mark here. In the future we need a method in structure operation that 
For existing Atom: update the attribute
For new Chain/Residue/Atom: add to Structure

Reply:
I agree. This is imperfect and there is a need for some more code to be added in structure operation. The issue I ran into is that it was not possible to use `update_structure()` from `structure_operation` since there will be new residues added.

Reply:
Yeah. I will file an issue for this

Reply:
Oh I saw the issue you made. Nice!